### PR TITLE
feat(funnels-correlation): simplify method for extracting property keys + values

### PR DESCRIPTION
**Pull request by yakkomajuri**
We do a whole bunch of stuff just to get a a 2d array with property keys and values as strings, but we can simplify it all by using `JSONExtractKeysAndValues` and force a string cast rather than using regex replace.

Proof this works as expected:

https://fiddle.clickhouse.com/da1ef638-79ba-47bf-9123-48506ce9e769

### CODE REVIEW

**Review by neilkakkar — APPROVED** [review]
awesome!

Yep, we were on an unsupported version before

File: ee/clickhouse/queries/funnels/test/__snapshots__/test_funnel_correlation.ambr [hunk 1]
```diff
@@ -3228,7 +3228,7 @@
   FROM
     (SELECT actor_id,
             funnel_actors.steps as steps,
-            arrayMap(x -> x.1, JSONExtractKeysAndValuesRaw(groups_0.group_properties_0)) as person_prop_keys,
+            JSONExtractKeys(groups_0.group_properties_0) as person_prop_keys,
```
Comment (neilkakkar) [inline ee/clickhouse/queries/funnels/test/__snapshots__/test_funnel_correlation.ambr:3231]: these snapshots should've had the `arrayJoin()` too no? 🤔

Comment (yakkomajuri) [reply ee/clickhouse/queries/funnels/test/__snapshots__/test_funnel_correlation.ambr:3231]: Literally committed the change 30sec before the review :D 

https://github.com/PostHog/posthog/pull/14031/commits/890c7d699d529cba2b6f777f12686292e590fb1b

Comment (yakkomajuri) [reply ee/clickhouse/queries/funnels/test/__snapshots__/test_funnel_correlation.ambr:3231]: Note that the scope changed from the original `JSONExtractKeys` change

**Pull request merged**